### PR TITLE
fix(mixin): ThanosSidecarUnhealthy doesn't fire if the sidecar is never healthy

### PR DESCRIPTION
**Pull request by arajkumar**
This PR reverts #3204 as it makes `ThanosSidecarUnhealthy` alert non functional by considering `thanos_sidecar_last_heartbeat_success_time_seconds` metric's timestamp instead of it's actual value for alerting. `timestamp(..)` function is associated with each scrape and regardless of the metric's value it's timestamp would be updated.

Consider the following scenario for `thanos_sidecar_last_heartbeat_success_time_seconds` where the heat-beat completed  only once.

t | t+1 | t+2 | t+3 
---|---|---|---
100 | 100 | 100 | 100

When applying `timestamp(thanos_sidecar_last_heartbeat_success_time_seconds)` on the above series values, it will always give us time at which scarp completed for this metrics.

In addition to above change, added a `for` clause in `ThanosSidecarUnhealthy` alert to improve it's usability, otherwise it would trigger immediately during sidecar boot-up.

Fixes https://github.com/thanos-io/thanos/issues/3990

### CODE REVIEW

**Review by dgrisonnet — APPROVED** [review]
lgtm

cc @bwplotka @onprem

File: mixin/alerts/sidecar.libsonnet [hunk 1]
```diff
@@ -45,8 +45,9 @@
               summary: 'Thanos Sidecar is unhealthy.',
             },
             expr: |||
-              time() - max by (%(dimensions)s) (timestamp(thanos_sidecar_last_heartbeat_success_time_seconds{%(selector)s})) >= 240
+              time() - max by (%(dimensions)s) (thanos_sidecar_last_heartbeat_success_time_seconds{%(selector)s}) >= 240
             ||| % thanos.sidecar,
+            'for': '5m',
```
Comment (onprem) [inline mixin/alerts/sidecar.libsonnet:50]: Small nit: let's remove quotes around `for`, just for consistency.

**Review by onprem — APPROVED** [review]
LGTM! Thanks. I just have a small non-blocking comment.

File: mixin/alerts/sidecar.libsonnet (hunk 1 above)
Comment (paulfantom) [reply mixin/alerts/sidecar.libsonnet:50]: `for` is a keyword in jsonnet and as such, it needs to be quoted.

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**